### PR TITLE
Bump version to 1.2.1

### PR DESCRIPTION
**Pull request by horis233**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Bump version 1.2.1
```

### CODE REVIEW

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+                    "name": "jenkins"
+                  }
+                ],
+                "registry": "example-service"
+              }
+            ]
+          }
+        }
+      ]
+	  capabilities: Seamless Upgrades
```
Comment (chenzhiwei) [inline deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:82]: remove these blank spaces.

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+            matchLabels:
+              name: operand-deployment-lifecycle-manager
+          strategy: {}
+          template:
+            metadata:
+              annotations:
+                productID: 068a62892a1e4db39641342e592daa25
+                productMetric: FREE
+                productName: IBM Cloud Platform Common Services
+                productVersion: 3.4.1
```
Comment (chenzhiwei) [inline deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:487]: The CS product version should be 3.4, right?

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: POD_NAMESPACE
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.namespace
+                - name: OPERATOR_NAME
+                  value: operand-deployment-lifecycle-manager
+                image: quay.io/opencloudio/odlm:1.2.1
```
Comment (chenzhiwei) [inline deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:525]: what's the difference between 1.2.1 and latest?

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,569 @@
...
+          - operator.ibm.com
+          - operators.coreos.com
+          resources:
+          - '*'
+          - operandregistries
+          - operandconfigs
+          - operandrequests
+          - operandbindinfos
+          verbs:
+          - '*'
```
Comment (chenzhiwei) [inline deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:461]: you can remove line 406~461, all these permissions are covered by following one.

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml (hunk 3 above)
Comment (horis233) [reply deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:525]: 1.2.1 is picked by CICD from the latest image. I will update it to use latest

File: deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml (hunk 2 above)
Comment (horis233) [reply deploy/olm-catalog/operand-deployment-lifecycle-manager/1.2.1/operand-deployment-lifecycle-manager.v1.2.1.clusterserviceversion.yaml:487]: I will change it back to `3.4.0`. I will ask Steve about which version should we use exactly.

**Comment by horis233** [comment]
@chenzhiwei Done, Please review again.

**Comment by chenzhiwei** [comment]
looks good. @DanielXLee Please double review.

**Pull request merged**